src/app/modules/training-program/components/programs/program-list.component.ts

```diff
@@ -1,27 +1,46 @@
-import { Component } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { map, catchError } from 'rxjs/operators';
+import { of as observableOf } from 'rxjs';
 
 import { Program } from 'src/app/interfaces/program';
 import { ProgramService} from '../../services/program.service';
-import { GenericListComponent } from 'src/app/generics/generic-list/generic-list';
 
 @Component({
   selector: 'app-program-list',
   templateUrl: './program-list.component.html',
   styleUrls: ['./program-list.component.css']
 })
-export class ProgramListComponent extends GenericListComponent<Program> {
+
+export class ProgramListComponent implements OnInit {
+
+  programs: Program[] = [];
+  /** The total number of programs. */
+  programsLength = 0;
+  /** Indicate data loading status */
+  isLoadingResults = true;
+
   constructor(
-    protected readonly route: ActivatedRoute,
-    protected readonly router: Router,
+    private readonly router: Router,
     private readonly programService: ProgramService,
-  ) {
-    super(route, router);
-  }
+  ) { }
 
+  ngOnInit() {
+    this.programService.getPrograms({offset: 0, limit: 100}).pipe(
+      map(data => {
+        this.isLoadingResults = false;
+        this.programsLength = data.count;
+        return data.results;
+      }),
+      catchError((err) => {
+        this.isLoadingResults = false;
+        return observableOf([]);
+      }),
+    ).subscribe(programs => this.programs = programs);
+  }
 
-  getResults(offset: number, limit: number) {
-    return this.programService.getPrograms({offset, limit});
+  navigateToDetail(row: Program) {
```

Comment:
The name of this function is different from what it does, rename it, `navigateToRelatedEvents` for example.